### ISSUE TITLE
Environment names are not unique

### ISSUE THREAD

**Issue by curusarn** [grafana/tanka#449]
When I create a new environment using `tk env add environments/prom-grafana/dev` the name of the environment is `dev`. That took me by surprise. I expected the name to be `prom-grafana/dev` because then the environment names would be guaranteed to be unique.

Why do I think the uniqueness of the environment names is important? 

When I set `injectLabels: true` then I could use the labels to see which Tanka environment a given k8s resource belongs to. I can't really do this if the labels are not unique.

Additionally, the labels are used by `tk prune` to identify resources which are missing from the local state. How does the `tk prune` deal with situations where multiple Tanka environments share the same name (plus what if they also share the same k8s namespace)? 

Am I missing something? Is there a reason to not use the full path as environment name? 

As an example, this is how I would get into a situation where I have multiple environments with the same name: 
- `tk env add environments/prom-grafana/dev`
- `tk env add environments/prom-grafana/prod`
- `tk env add environments/something-else/dev`
- `tk env add environments/something-else/prod`

TL;DR
Why is the Tanka environment name just the last directory instead of a full path to the environment? Using a full path as the environment name would make the environment names unique.

**Comment by Duologic** [comment]
The name set in spec.json is kind of ignored, the name will be set to the path relative to tanka root at runtime. This is indeed confusing.

**Comment by Duologic** [comment]
Not this is not true for inline environments, however, since v0.13.0 there also is the 'metadata.namespace' keyword, which is set to the path relative to the tanka root, similar to Kubernetes, the name+namespace would then guarantee uniqueness.

Also see #433 in relation to pruning.
